### PR TITLE
nRF Desktop: SUIT semantic support 

### PR DESCRIPTION
**Pull request by kapi-no**
- [x] Test the code with the SDFW that supports semantic versioning

### CODE REVIEW

**Comment by kapi-no** [comment]
Rebased on top of the main branch

File: applications/nrf_desktop/src/modules/dfu.c [hunk 1]
```diff
@@ -829,6 +845,21 @@ static void handle_image_info_request(uint8_t *data, size_t *size)
 		uint8_t minor = 0;
 		uint16_t revision = 0;
 
+		if (is_semver_supported) {
+			LOG_INF("Booted application version: %d.%d.%d+%d",
```
Comment (tomchy) [inline applications/nrf_desktop/src/modules/dfu.c:849]: The `+%d` notation may be misleading as SUIT does not provide build number.
The notations is `%d.%d.%d-%s.%d` where `%s` is the pre-release name.

**Review by tomchy — APPROVED** [review]

File: applications/nrf_desktop/src/modules/dfu.c (hunk 1 above)
Comment (kapi-no) [reply applications/nrf_desktop/src/modules/dfu.c]: improved this behaviour

**Review by mkapala-nordic — APPROVED** [review]

**Comment by kapi-no** [comment]
Small fix for the `if` condition in the Python script to suppress update of the DFU image with the semantic version support when the booted firmware uses the semantic version feature

**Review by alstrzebonski — APPROVED** [review]

File: applications/nrf_desktop/doc/dfu.rst [hunk 2]
```diff
@@ -142,7 +142,12 @@ fwinfo
    For the nRF54H Series:
```
Comment (annwoj) [inline applications/nrf_desktop/doc/dfu.rst:142]:
```suggestion
   Additionally, for the nRF54H Series, the following applies:
```

File: applications/nrf_desktop/doc/dfu.rst [hunk 3]
```diff
@@ -142,7 +142,12 @@ fwinfo
...
 
    * Reported image size is set to zero.
```
Comment (annwoj) [inline applications/nrf_desktop/doc/dfu.rst:144]:
```suggestion
   * The reported image size is set to zero.
```

File: applications/nrf_desktop/doc/dfu.rst [hunk 4]
```diff
@@ -142,7 +142,12 @@ fwinfo
...
-   * Root manifest sequence number is reported as the booted image version.
+   * The booted image version is indicated using:
+
+     * Root manifest sequence number that is encoded in the build number field.
+     * Manifest semantic version if the SDFW supports semantic versioning (requires v0.6.2 and higher).
```
Comment (annwoj) [inline applications/nrf_desktop/doc/dfu.rst:148]:
```suggestion
     * Manifest semantic version, if supported by the SDFW (requires v0.6.2 or higher).
```

File: applications/nrf_desktop/doc/dfu.rst [hunk 5]
```diff
@@ -142,7 +142,12 @@ fwinfo
...
+       Semantic version is encoded in the major, minor and patch fields.
```
Comment (annwoj) [inline applications/nrf_desktop/doc/dfu.rst:149]:
```suggestion
       The semantic version is encoded in the major, minor and patch fields.
```

File: applications/nrf_desktop/doc/dfu.rst [hunk 6]
```diff
@@ -142,7 +142,12 @@ fwinfo
    For the nRF54H Series:
 
    * Reported image size is set to zero.
-   * Root manifest sequence number is reported as the booted image version.
+   * The booted image version is indicated using:
```
Comment (annwoj) [inline applications/nrf_desktop/doc/dfu.rst:145]:
```suggestion
   * The booted image version is indicated by:
```

**Review by annwoj — APPROVED** [review]

**Pull request merged**